Stripe/PublicHeaders/STPCustomerContext.h

```diff
@@ -0,0 +1,48 @@
+//
+//  STPCustomerContext.h
+//  Stripe
+//
+//  Created by Ben Guo on 5/2/17.
+//  Copyright © 2017 Stripe, Inc. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+
+#import "STPBackendAPIAdapter.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@protocol STPEphemeralKeyProvider;
+@class STPEphemeralKey, STPEphemeralKeyManager;
+
+/**
+ An `STPCustomerContext` retrieves and updates a Stripe customer using
+ an ephemeral key, a short-lived API key scoped to a specific Customer object.
+ */
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated"
+@interface STPCustomerContext : NSObject <STPBackendAPIAdapter>
+#pragma clang diagnostic pop
+
+/**
+ Whenever the customer context retrieves a customer, it will return a cached
+ value if it was retrieved less than this number of seconds ago.
+ The default value is 60 seconds.
+ */
+@property (nonatomic, assign) NSTimeInterval cachedCustomerMaxAge;
```

Comment:
My thinking exposing it was that it makes it clearer that CustomerContext does caching, which might be surprising. After thinking about it some more, I think it's probably better to hide it until someone asks for the configuration option in the future. It's hard to come up with a good recommendation for how to come up with an appropriate value if you want to change it, since when CustomerContext retrieves a customer is opaque to the developer and out of their control.